client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -55,31 +220,18 @@ export const UpgradePlanDetails = ( props: Props ) => {
 
 	return (
 		<div className="import__upgrade-plan-details">
-			<QuerySitePlans siteId={ siteId } />
-			<div className="import__upgrade-plan-period-switcher">
-				<ButtonGroup>
-					<Button
-						borderless
-						className={ clsx( { selected: selectedPlan === PLAN_BUSINESS_MONTHLY } ) }
-						onClick={ () => setSelectedPlan( PLAN_BUSINESS_MONTHLY ) }
-					>
-						{ __( 'Pay monthly' ) }
-					</Button>
-					<Button
-						borderless
-						className={ clsx( { selected: selectedPlan === PLAN_BUSINESS } ) }
-						onClick={ () => setSelectedPlan( PLAN_BUSINESS ) }
-					>
-						{ __( 'Pay annually' ) }
-					</Button>
-				</ButtonGroup>
-			</div>
+			<UpgradePlanPeriodSwitcher
+				introOfferAvailable={ introOfferAvailable }
+				selectedPlan={ selectedPlan }
+				onMonthlyPlanClick={ () => setSelectedPlan( PLAN_BUSINESS_MONTHLY ) }
+				onAnnualPlanClick={ () => setSelectedPlan( PLAN_BUSINESS ) }
+			/>
```

Comment:
With the change in the `UpgradePlanPeriodSwitcher` to toggle display via `introOfferAvailable`, we could simply inline that check. We could even use the old `div` from before your changes with `introOfferAvailable` as a guard.

I don't have a strong opinion, but I have a slight preference for the local guard so it's easier to understand from the main component when the toggle will (not) be shown.

```suggestion
			{ introOfferAvailable && (
				<UpgradePlanPeriodSwitcher
					introOfferAvailable={ introOfferAvailable }
					selectedPlan={ selectedPlan }
					onMonthlyPlanClick={ () => setSelectedPlan( PLAN_BUSINESS_MONTHLY ) }
					onAnnualPlanClick={ () => setSelectedPlan( PLAN_BUSINESS ) }
				/>
			)
		}
```